### PR TITLE
(fix) revise logic for saving attachments

### PR DESCRIPTION
**Pull request by arodidev**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR seeks to ensure that no request is made to the attachments endpoint if no field bearing an attachment was filled at the point of submission.

### CODE REVIEW

File: src/components/encounter/encounter-form.component.tsx [hunk 1]
```diff
@@ -472,26 +472,30 @@ const EncounterForm: React.FC<EncounterFormProps> = ({
...
-        const attachmentsResponse = await Promise.all(
-          EncounterFormManager.saveAttachments(fields, savedEncounter, abortController),
-        );
-        if (attachmentsResponse?.length) {
-          showSnackbar({
-            title: t('attachmentsSaved', 'Attachment(s) saved successfully'),
-            kind: 'success',
-            isLowContrast: true,
+
+      if (fields.filter((eachField) => eachField.questionOptions.rendering === 'file')?.length) {
```
Comment (samuelmale) [inline src/components/encounter/encounter-form.component.tsx:476]: Probably a better place to handle this is within the encounter form manager.

File: src/components/encounter/encounter-form.component.tsx [hunk 2]
```diff
@@ -465,26 +465,30 @@ const EncounterForm: React.FC<EncounterFormProps> = ({
...
-        if (attachmentsResponse?.length) {
-          showSnackbar({
-            title: t('attachmentsSaved', 'Attachment(s) saved successfully'),
-            kind: 'success',
-            isLowContrast: true,
+
+      if (EncounterFormManager.evaluatedAttachmentFields(fields)?.length) {
+        try {
+          const attachmentsResponse = await Promise.all(
+            EncounterFormManager.saveAttachments(fields, savedEncounter, abortController),
```
Comment (samuelmale) [inline src/components/encounter/encounter-form.component.tsx:472]: Can you just make sure that if there are zero attachments this function returns an empty value for `attachmentsResponse` and implement the necessary filters internally?

Comment (arodidev) [reply src/components/encounter/encounter-form.component.tsx:472]: You mean something like:

```ts
const evaluatedFields = EncounterFormManager.evaluatedAttachmentFields(fields);
if (evaluatedFields?.length) {
  try {
    const attachmentsResponse = await Promise.all(
      EncounterFormManager.saveAttachments(fields, savedEncounter, abortController),
    );
    if (attachmentsResponse?.length) {
      showSnackbar({
        title: t('attachmentsSaved', 'Attachment(s) saved successfully'),
        kind: 'success',
        isLowContrast: true,
      });
    }
  } catch (error) {
    const errorMessages = extractErrorMessagesFromResponse(error);
    return Promise.reject({
      title: t('errorSavingAttachments', 'Error saving attachment(s)'),
      subtitle: errorMessages.join(', '),
      kind: 'error',
      isLowContrast: false,
    });
  }
} else {
  return [];
}

```

Comment (CynthiaKamau) [reply src/components/encounter/encounter-form.component.tsx:472]: I think the logic can be handled in the `EncounterFormManager.saveAttachments` function, 

```
  static saveAttachments(fields: FormField[], encounter: OpenmrsEncounter, abortController: AbortController) {
    const complexFields = fields?.filter((field) => field?.questionOptions.rendering === 'file');
    return complexFields.map((field) => {
      const patientUuid = typeof encounter?.patient === 'string' ? encounter?.patient : encounter?.patient?.uuid;
      return saveAttachment(
        patientUuid,
        field,
        field?.questionOptions.concept,
        new Date().toISOString(),
        encounter?.uuid,
        abortController,
      );
    });
  }
```

Somewhere in here, before` return saveAttachment(`

Comment (arodidev) [reply src/components/encounter/encounter-form.component.tsx:472]: Thanks for the clarification @CynthiaKamau

**Review by CynthiaKamau — APPROVED** [review]

**Pull request merged**